src/SSVOpenHexagon/Core/main.cpp

```diff
@@ -42,11 +36,15 @@ int main(int argc, char* argv[])
     hg::Online::setCurrentGtm(
         std::make_unique<hg::Online::GlobalThreadManager>());
 
-    const auto overrideIds = [&] {
+    const auto overrideIds = [&]
+    {
         std::vector<std::string> result;
         for(int i{0}; i < argc; ++i)
         {
-            result.emplace_back(argv[i]);
+            if(!strcmp(argv[i], "-l")) // need this later
+                i += 2;
```

Comment:
Rather than looping again afterwards, you could store the found level here, maybe into a `std::optional<std::string>` so that you can also check later whether any `-l` flag was supplied or not. Happy to elaborate on Discord.